### PR TITLE
feat: add seed function

### LINKED ISSUE #8: Seeded random number generator instead of Math.random
### Description

Right now `falso` generator is nondeterministic.
Using it in some snapshot tests will make tests always fail.

### Proposed solution

Add a way to set a starting seed number for random number generator

Proposed API

```ts
import { seed, randomNumber } from 'falso'

seed('some-seed') // set seed to some constant
randomNumber() // always returns same number
```

https://github.com/davidbau/seedrandom
https://unpkg.com/browse/faker@5.5.3/lib/random.js

### Alternatives considered

_No response_

### Do you want to create a pull request?

Yes

### PR DESCRIPTION
**Pull request by iamandrewluca**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngneat/falso/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[x] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Right now falso generator is nondeterministic.
Using it in some snapshot tests will make tests always fail.

Issue Number: #8

## What is the new behavior?

This PR allows to set a constant seed to get same results for same seed value

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

Closes #8

### CODE REVIEW

**Comment by iamandrewluca** [comment]
@NetanelBasal ping me if any changes are not needed. README file was modified by prettier

File: package.json [hunk 1]
```diff
@@ -53,4 +55,4 @@
   "lint-staged": {
     "*.{ts,json,html,tsx}": "prettier --write"
```
Comment (iamandrewluca) [inline package.json:56]: here I think we should also include markdown files, users forget to keep same stlye in README.md

**Comment by iamandrewluca** [comment]
I think `seedrandom` dependency should be in another place 🤔

File: package.json [hunk 2]
```diff
@@ -37,6 +38,7 @@
     "jest": "27.2.3",
     "lint-staged": "11.1.2",
     "prettier": "2.3.1",
+    "seedrandom": "^3.0.5",
```
Comment (NetanelBasal) [inline package.json:41]: lock the version, please.

File: packages/falso/src/tests/seed.spec.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import { seed } from '../lib/random';
+
+describe('seed', () => {
+  it('should return same values with constant seed', () => {
+    const seed1Value = 'constant-seed-1';
+    const seed2Value = 'constant-seed-2';
+    const values = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+    seed(seed1Value);
+    expect(rand(values)).toEqual(1);
```
Comment (NetanelBasal) [inline packages/falso/src/tests/seed.spec.ts:11]: How do you know that value you're going to get?

File: packages/falso/src/tests/seed.spec.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    seed(seed2Value);
+    expect(rand(values)).toEqual(7);
+    expect(rand(values)).toEqual(5);
+    expect(rand(values)).toEqual(1);
+
+    seed(seed1Value);
+    expect(rand(values)).toEqual(1);
+    expect(rand(values)).toEqual(6);
+    expect(rand(values)).toEqual(5);
+  });
```
Comment (NetanelBasal) [inline packages/falso/src/tests/seed.spec.ts:24]: Can you please add a test for resetting back?

File: packages/falso/src/tests/seed.spec.ts (hunk 3 above)
Comment (iamandrewluca) [reply packages/falso/src/tests/seed.spec.ts:11]: I run it first time, and saw the value. Running the function with same seed should return same value

File: package.json (hunk 2 above)
Comment (iamandrewluca) [reply package.json:41]: 🤣 ok

File: packages/falso/src/tests/seed.spec.ts (hunk 4 above)
Comment (iamandrewluca) [reply packages/falso/src/tests/seed.spec.ts:24]: I wanted, but I don't know what exactly I want to test 🤔, because `rand` is nondeterministic again

Comment (NetanelBasal) [reply packages/falso/src/tests/seed.spec.ts:24]: Yes

File: packages/falso/src/tests/seed.spec.ts (hunk 3 above)
Comment (NetanelBasal) [reply packages/falso/src/tests/seed.spec.ts:11]: Are we using the fastest algorithm they provide?

**Comment by NetanelBasal** [comment]
https://github.com/ngneat/falso/pull/12#issuecomment-1009424415

Not sure why.

**Comment by iamandrewluca** [comment]
@NetanelBasal I trid to make a build of the library locally, and `seedrandom` is missing from `package.json` 🤔

**Comment by NetanelBasal** [comment]
It should be a dependency of the library. Add it to `package.json` of the library.

**Comment by iamandrewluca** [comment]
Hmmm, moved it in `packages/falso/package.json`, but nx build fails after this 🤔 

image

**Comment by iamandrewluca** [comment]
nevermind fixed

File: package.json [hunk 5]
```diff
@@ -8,6 +8,9 @@
     "prepare": "husky install",
     "test": "echo 'test'"
   },
+  "workspaces": [
+    "packages/falso"
+  ],
```
Comment (iamandrewluca) [inline package.json:13]: this is a npm@7+ feature, on `npm install` in root will automatically install also packages for `packages/falso`

File: packages/falso/.gitignore [hunk 6]
```diff
@@ -0,0 +1,130 @@
+# Logs
```
Comment (iamandrewluca) [inline packages/falso/.gitignore:1]: Generated with
```shell
npx gitignore Node
```

File: packages/falso/package.json [hunk 7]
```diff
@@ -26,5 +26,9 @@
   "repository": {
     "type": "git",
     "url": "https://github.com/ngneat/falso"
+  },
+  "dependencies": {
+    "@types/seedrandom": "3.0.1",
+    "seedrandom": "3.0.5"
```
Comment (iamandrewluca) [inline packages/falso/package.json:32]: now final build includes these dependencies

File: packages/falso/src/tests/seed.spec.ts (hunk 3 above)
Comment (iamandrewluca) [reply packages/falso/src/tests/seed.spec.ts:11]: From `seedrandom` docs, I see that it says that `alea` is fastest 🤔 

https://github.com/davidbau/seedrandom#other-fast-prng-algorithms

File: packages/falso/.gitignore (hunk 6 above)
Comment (NetanelBasal) [reply packages/falso/.gitignore:1]: why we need this? we have one in the root.

File: packages/falso/package.json (hunk 7 above)
Comment (NetanelBasal) [reply packages/falso/package.json:32]: @types/seedrandom should be dev dep.

File: packages/falso/src/tests/seed.spec.ts (hunk 3 above)
Comment (NetanelBasal) [reply packages/falso/src/tests/seed.spec.ts:11]: Hmm, we can start with the current one.

File: packages/falso/.gitignore (hunk 6 above)
Comment (iamandrewluca) [reply packages/falso/.gitignore:1]: One in the root ignores only `node_modules` from root.
I can change `.gitignore` from root to ignore all `node_modules`, should I?

**Pull request merged**